packages/loot-core/src/types/api-handlers.ts

```diff
@@ -186,4 +194,28 @@ export interface ApiHandlers {
   'api/rule-update': (arg: { rule: RuleEntity }) => Promise<RuleEntity>;
 
   'api/rule-delete': (id: string) => Promise<boolean>;
+
+  'api/schedule-create': (arg: {
+    schedule: Partial<ScheduleEntity>;
+    conditions: Partial<RuleConditionEntity>[];
+  }) => Promise<string>;
+
+  'api/schedule-update': (arg: {
+    id: string;
+    conditions: Partial<RuleConditionEntity>[];
+    resetNextDate?: boolean;
+  }) => Promise<string>;
+
+  'api/schedule-delete': (id: string) => Promise<void>;
+
+  'api/schedule-skip-next-date': (id: string) => Promise<void>;
+
+  'api/schedule-post-transaction': (id: string) => Promise<void>;
+
+  'api/schedule-get-upcoming-dates': (arg: {
+    config: RecurConfig;
+    count: number;
+  }) => Promise<string[]>;
```

Comment:
Not sure if we can use DateLike here as return type or introduce a new type e.g. DateString. But feel free to ignore here, it can be done on another PR.
```suggestion
  }) => Promise<DateString[]>;
```

Reply:
Yeah... I mean you're probably right. I actually want to go through and add proper typings to the rest of the API methods, but I don't want this PR's scope to grow any larger. I may add a DateString type as part of that PR